packages/squiggle-lang/src/fr/list.ts

```diff
@@ -120,6 +132,28 @@ export const library = [
       ),
     ],
   }),
+  maker.make({
+    name: "uniq",
+    requiresNamespace: true,
+    examples: [`List.uniq([1,2,3,"hi",false,"hi"])`],
+    definitions: [
+      makeDefinition<Value[]>("uniq", [frArray(frAny)], ([arr]) => {
+        const isUniqableType = (t: Value) =>
+          includes(["String", "Bool", "Number"], t.type);
+        //I'm not sure if the r.type concat is essential, but seems safe.
+        const uniqueValueKey = (t: Value) => t.toString() + t.type;
+
+        const allUniqable = arr.every(isUniqableType);
+        if (allUniqable) {
+          return Ok(vArray(uniqBy(arr, uniqueValueKey)));
+        } else {
+          return Result.Error(
+            REOther("Can only apply uniq() to Strings, Numbers, or Bools")
+          );
+        }
+      }),
```

Comment:
How about something like this instead?

```typescript
let seen = new Set();
let res: Value[] = [];
for (let value of arr) {
  if (value.type === "Void" || seen.has(value.value)) {
    continue;
  }
  seen.add(value.value);
  res.push(value);
}
return Ok(vArray(res));
```

(I haven't checked if it works)

Or maybe it should fail on Voids, I'm not sure about that. But this version should work for all other value types.

Reply:
How would it tell if distributions are the same? Like, two pointset Normal(5,2) distributions? Similar with lamdas?

Reply:
By object identity. It works fine with sets. Two uniquely constructed identical dists would be different, but two references to the same dist would be de-duplicated.

It works the same way with `_.uniq`, btw (so you could do it with lodash, but for function implementations I'd generally prefer to write the code by hand, so that we know what it's doing and there are no surprises):

```
> x = {foo: 5}
{ foo: 5 }
> y = {foo: 5}
{ foo: 5 }
> s = new Set()
Set(0) {}
> s.add(x)
Set(1) { { foo: 5 } }
> s.has(x)
true
> s.has(y)
false
> s.add(y)
Set(2) { { foo: 5 }, { foo: 5 } }
> s.has(y)
true
> s.add(y)
Set(2) { { foo: 5 }, { foo: 5 } }
> > _.uniq([4,5,6,5,x,x,y,y,y])
[ 4, 5, 6, { foo: 5 }, { foo: 5 } ]
```

Reply:
>  Two separately constructed identical dists would be different

That's my problem. As a user, I'd expect that ``uniq`` would remove one of these. I think this will be confusing, so want to hold it off until later on. For now, we can promise the correct functionality with strings/numbers/bools, which is still useful.

Reply:
I see. Yes, it's fine to hold off those decisions until later, my point was more about doing imperative implementation instead of relying on string concatenation, which seems somewhat hacky.

But it doesn't matter much now, and can be refactored later, I don't want to delay the merge further with this.